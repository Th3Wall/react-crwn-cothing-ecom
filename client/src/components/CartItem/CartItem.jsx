import { memo } from 'react';
import {
    CartItemDetails,
    CartItemImage,
    CartItemName,
    CartItemPrice,
    CartItemWrp
} from './cartItem.styles';

const CartItem = ({ item: { imageUrl, name, price, quantity }, variants}) => {
    return (
        <CartItemWrp
            variants={variants}
        >
            <CartItemImage src={imageUrl} alt="cart-item"/>
            <CartItemDetails>
                <CartItemName>{name}</CartItemName>
                <CartItemPrice>{quantity} x ${price}</CartItemPrice>
            </CartItemDetails>
        </CartItemWrp>
    )
}

export default memo(CartItem);
