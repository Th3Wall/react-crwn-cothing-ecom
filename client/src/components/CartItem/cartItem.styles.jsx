import styled, { css } from "styled-components";
import { motion } from 'framer-motion';

const cartItemDetailStyles = css`
    font-size: 16px;
`

export const CartItemWrp = styled(motion.div)`
    width: 100%;
    display: flex;
    height: auto;
    margin-bottom: 15px;
    background-color: #f3f3f3;
    border-radius: 12px;
    padding: 10px;
`

export const CartItemImage = styled.img`
    width: 100%;
    border-radius: 12px;
    max-width: 20%;
    height: 85px;
`

export const CartItemDetails = styled.div`
    width: 70%;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: center;
    padding: 10px 20px;
`

export const CartItemName = styled.span`
    ${cartItemDetailStyles}
`
export const CartItemPrice = styled.span`
    ${cartItemDetailStyles}
`