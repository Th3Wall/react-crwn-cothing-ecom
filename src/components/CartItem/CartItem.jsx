import { CartItemDetails, CartItemImage, CartItemName, CartItemPrice, CartItemWrp } from './cartItem.styles';

const CartItem = ({ item: { imageUrl, name, price, quantity }}) => {
    return (
        <CartItemWrp>
            <CartItemImage src={imageUrl} alt="cart-item"/>
            <CartItemDetails>
                <CartItemName>{name}</CartItemName>
                <CartItemPrice>{quantity} x ${price}</CartItemPrice>
            </CartItemDetails>
        </CartItemWrp>
    )
}

export default CartItem;
