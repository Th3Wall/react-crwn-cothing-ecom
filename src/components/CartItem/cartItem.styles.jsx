import styled, { css } from "styled-components";

const cartItemDetailStyles = css`
    font-size: 16px;
`

export const CartItemWrp = styled.div`
    width: 100%;
    display: flex;
    height: 80px;
    margin-bottom: 15px;
`

export const CartItemImage = styled.img`
    width: 30%;
    border-radius: 12px;
`

export const CartItemDetails = styled.div`
    width: 70%;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: center;
    padding: 10px 20px;
`

export const CartItemName = styled.span`
    ${cartItemDetailStyles}
`
export const CartItemPrice = styled.span`
    ${cartItemDetailStyles}
`