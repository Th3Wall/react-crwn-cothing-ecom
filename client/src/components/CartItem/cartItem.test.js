import React from 'react';
import { shallow } from 'enzyme';
import CartItem from './CartItem';

describe('CartItem', () => {
    it('should display CartItem component', () => {

        const mockProperties = {
            item: {
                imageUrl: 'https://i.ibb.co/1RcFPk0/white-nike-high-tops.png',
                name: 'Sneakers Nike Air Max',
                price: 99,
                quantity: 1
            }
        }

        const wrapper = shallow(<CartItem {...mockProperties} />);
        expect(wrapper).toMatchSnapshot();
    })
})