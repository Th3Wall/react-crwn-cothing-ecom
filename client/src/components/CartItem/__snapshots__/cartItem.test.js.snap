// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem should display CartItem component 1`] = `
<Styled(Component)>
  <styled.img
    alt="cart-item"
    src="https://i.ibb.co/1RcFPk0/white-nike-high-tops.png"
  />
  <styled.div>
    <styled.span>
      Sneakers Nike Air Max
    </styled.span>
    <styled.span>
      1
       x $
      99
    </styled.span>
  </styled.div>
</Styled(Component)>
`;
